tools/logbucket-global-to-regional/main.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python
+#
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+    Currently when logging API is enabled in a project it creates
+    two log buckets: _Required and _Default.These log buckets are
+    created in a Global region, and customers with data locality restriction
+    may not want to store their logs in a Global location.
+    https://cloud.google.com/logging/docs/routing/overview#buckets
+
+    This behaviour can be changed by setting an organization level setting
+    that restricts the log bucket creation to a specefic region.
+    However, these settings are applied only for any newly created projects
+    and doesn't change the log bucket location for existing projects.
+    https://cloud.google.com/logging/docs/default-settings
+
+    In some cases, customers may not have set these organization level setting
+    and would want to change the log bucket's location to a specific region.
+
+    P.S: _Required log bucket region can NOT be changed once they are created.
+
+    This utility helps in reconfiguring the _Default sink to send logs to a
+    newly created regional Default bucket.
+
+    There are 2 parts to this utility:
+        1. list_all_projects --> This function takes an Organization ID and
+            scans through all the folders and sub folders to generate a list
+            of projects within the organization
+        2. create_bucket_update_sink --> This function takes a list of projects
+            and reconfigures the _Default sink
+
+    If you already have a list of projects where you want to reconfigure the
+    _Default sink, you can disable project listing by setting "list_projects = False".
+
+    Permissions and Roles Required: The principal running the utility needs
+    the below roles attached to them at the organization level.
+        1. roles/resourcemanager.organizationViewer
+        2. roles/resourcemanager.folderViewer
+        3. roles/logging.admin
+"""
+
+from collections import deque
+import logging
+from pathlib import Path
+import time
+import errno
+import os
+from google.cloud import resourcemanager_v3
+from google.cloud.logging_v2.types import logging_config
+from google.cloud.logging_v2.services.config_service_v2 import ConfigServiceV2Client
+
+# Logger configurations
+logger = logging.getLogger(__name__)
+logger.setLevel(level=logging.INFO)
```

Comment:
u can keep a TO DO so that users can change this to debug for viuewing DEBUG logs